qiskit/transpiler/target.py

```diff
@@ -0,0 +1,600 @@
+    """
+
+    __slots__ = (
+        "num_qubits",
+        "_gate_map",
+        "_gate_name_map",
+        "_qarg_gate_map",
+        "description",
+        "_coupling_graph",
+        "_instruction_durations",
+        "_instruction_schedule_map",
+        "dt",
+        "granularity",
+        "min_length",
+        "pulse_alignment",
+        "aquire_alignment",
+    )
+
+    def __init__(
+        self,
+        description=None,
+        num_qubits=0,
+        dt=None,
+        granularity=1,
+        min_length=1,
+        pulse_alignment=1,
+        aquire_alignment=1,
+    ):
+        """
+        Create a new Target object
+
+        Args:
+            description (str): An optional string to describe the Target.
+            num_qubits (int): An optional int to specify the number of qubits
+                the backend target has. If not set it will beimplicitly set
+                based on the qargs when :meth:`~qiskit.Target.add_instruction`
+                is called. Note this must be set if the backend target is for a
+                noiseless simulator that doesn't have constraints on the
+                instructions so the transpiler knows how many qubits are
+                available.
+            dt (float): The system time resolution of input signals in seconds
+            granularity (int): An integer value representing minimum pulse gate
+                resolution in units of ``dt``. A user-defined pulse gate should
+                have duration of a multiple of this granularity value.
+            min_length (int): An integer value representing minimum pulse gate
+                length in units of ``dt``. A user-defined pulse gate should be
+                longer than this length.
+            pulse_alignment (int): An integer value representing a time
+                resolution of gate instruction starting time. Gate instruction
+                should start at time which is a multiple of the alignment
+                value.
+            acquire_alignment (int): An integer value representing a time
+                resolution of measure instruction starting time. Measure
+                instruction should start at time which is a multiple of the
+                alignment value.
+        """
+        self.num_qubits = num_qubits
+        # A mapping of gate name -> gate instance
+        self._gate_name_map = {}
+        # A nested mapping of gate name -> qargs -> properties
+        self._gate_map = {}
+        # A mapping of qarg -> set(gate name)
+        self._qarg_gate_map = defaultdict(set)
+        self.dt = dt
+        self.description = description
+        self._coupling_graph = None
+        self._instruction_durations = None
+        self._instruction_schedule_map = None
+        self.granularity = granularity
+        self.min_length = min_length
+        self.pulse_alignment = pulse_alignment
+        self.aquire_alignment = aquire_alignment
+
```

Comment:
Hmm, that's weird there's nothing really different about target vs backend in this regard. I just checked my local interpreter and tab completion on a target object works fine. Are you having issues with `backend.target` `<tab>` or just `Target()` `<tab>`?. I could see that the former might have issues because it's missing the type hint (which I couldn't add because of circular import issues).

Reply:
Give it a try with: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/62f36bdccc0ab89d95e4207ff32fc88dd2320e20 instead of using type hints which I couldn't figure out a pattern that would work without a cyclical import I used the old docstring syntax to specify the return type and this seems to work for `backend.target`.

Reply:
it works now thanks.